### PR TITLE
2.2.17

### PR DESCRIPTION
**Pull request by j0be**
This corrects the click handling with the new default popup in the manifest.

### CODE REVIEW

File: extension/chrome/src/js/background.js [hunk 1]
```diff
@@ -174,8 +174,8 @@ function retrieveActivities() {
 
 		Promise.all([reviewersPromised, authorPromised]).then(function([reviewersResult, authorResult]) {
 			if (reviewersResult.errors) { throw reviewersResult.errors }
-			if (authorResult.errors) { throw authorResult.errors }
-			const allPR = reviewersResult.concat(authorResult)
+            if (authorResult.errors) { throw authorResult.errors }
```
Comment (ra100) [inline extension/chrome/src/js/background.js:177]: Can you, please, fix the indents, for better readability? Maybe caused by tabs vs spaces.

File: extension/chrome/src/js/userhosts.js [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 		replace(/;/g, ' ').
-		replace(/\r/g, ' ').
-		replace(/\t/g, ' ').
-		replace(/\n/g, ' ').
-		split(' ');
-
-	let fhosts = [];
+		replace(/\t+/g, ' ').
+		replace(/\r+/g, ' ').
+        replace(/\n+/g, ' ').
```
Comment (ra100) [inline extension/chrome/src/js/userhosts.js:9]: Can this be changed to just one replace?
```js
.replace(/[\n\t,;\s\r]+/g, ' ');
```

File: extension/chrome/src/js/userhosts.js [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
-		if (host != "") {
-			fhosts.push(host)
-		}
-	}
-	hosts = fhosts
 	if (hosts.length > 0) {
 		chrome.storage.sync.get('user_defined_hosts', oldhosts => {
-			chrome.storage.sync.set({'user_defined_hosts': (oldhosts.user_defined_hosts || []).concat(hosts)})
-			chrome.extension.getBackgroundPage().handleUserDefinedHosts()
+            let uniqueHosts = [...new Set(hosts)];
```
Comment (ra100) [inline extension/chrome/src/js/userhosts.js:18]: :+1:

File: extension/chrome/src/js/userhosts.js [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
-		replace(/\n/g, ' ').
-		split(' ');
-
-	let fhosts = [];
+		replace(/\t+/g, ' ').
+		replace(/\r+/g, ' ').
+        replace(/\n+/g, ' ').
+        trim().
+        split(' ').
+        filter(host => {
```
Comment (ra100) [inline extension/chrome/src/js/userhosts.js:12]: `.filter(Boolean)` could be used.
or `.filter(host => !!host)`, no need to wrap it in {}

**Review by ra100 — COMMENTED** [review]
Looks good, just some fomatting would be nice.

File: extension/chrome/src/js/background.js (hunk 1 above)
Comment (j0be) [reply extension/chrome/src/js/background.js:177]: Oh yep. I'm spaces. I'll correct it.

File: extension/chrome/src/js/userhosts.js (hunk 2 above)
Comment (j0be) [reply extension/chrome/src/js/userhosts.js:9]: Yep! I added the `+` so it grabs any recurring dividers.

**Comment by j0be** [comment]
@ra100, I added all code review corrections you mentioned

File: extension/chrome/src/js/userhosts.js [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
+			chrome.extension.getBackgroundPage().handleUserDefinedHosts();
 		});
-		hostsButton.value = ''	
 	}
-})
\ No newline at end of file
+});
+
+populateTextArea();
+chrome.extension.getBackgroundPage().handleExtensionClick();
```
Comment (j0be) [inline extension/chrome/src/js/userhosts.js:21]: This is what actually opens the tab from within the popup that has been added.

File: extension/chrome/src/js/background.js [hunk 6]
```diff
@@ -247,11 +247,15 @@ function pingAllExistingTabs() {
 
 
 // Click on extension icon
-chrome.browserAction.onClicked.addListener(function() {
+chrome.browserAction.onClicked.addListener(handleExtensionClick);
+
+function handleExtensionClick() {
 	chrome.storage.local.get(['currentStashBaseUrl'], function(items) {
-		chrome.tabs.create({'url': `${items.currentStashBaseUrl}/`});
+		if (items.currentStashBaseUrl) {
```
Comment (j0be) [inline extension/chrome/src/js/background.js:254]: If there is no `currentStashBaseUrl` it won't open a tab for it, which will leave the popup open

**Comment by dragouf** [comment]
@j0be @ra100 is it ok to merge ?

**Comment by ra100** [comment]
@dragouf it fixes the hosts issue, so yes for me.

**Pull request merged**